custom_components/candy/client/model.py

```diff
@@ -181,28 +185,279 @@ class OvenState(StatusCode):
+        return cls(
+            heater_state=HobHeaterStatus.parse_state(
+                json["Z" + index + "status"], json["Z" + index + "hot"]
+            ),
+            status=json["Z" + index + "status"] == "1",
+            pan=json["Z" + index + "pan"] == "1",
+            hot=json["Z" + index + "hot"] == "1",
+            low=json["Z" + index + "low"] == "1",
+            combi=json["Z" + index + "combi"] == "1",
+            power=int(json["Z" + index + "power"]),
+            timer_minutes=int(json["Z" + index + "timeh"]) * 60
+            + int(json["Z" + index + "timem"])
+            if json["Z" + index + "status"] == "1"
+            and (
+                int(json["Z" + index + "timeh"]) > 0
+                or int(json["Z" + index + "timem"]) > 0
+            )
+            else None,
+        )
+
+    @staticmethod
+    def parse_state(status, hot) -> str:
+        """
+        Parse Hob status
+        """
+        if status == "1":
+            return HobState.HEATING
+
+        if hot == "1":
+            return HobState.COOLING_DOWN
+        else:
+            return HobState.IDLE
+
+
+@dataclass
+class HobStatus:
+    machine_state: HobState
+    heater1: Optional[HobHeaterStatus]
+    heater2: Optional[HobHeaterStatus]
+    heater3: Optional[HobHeaterStatus]
+    heater4: Optional[HobHeaterStatus]
+    lock: bool
+    remote_control: bool
+
+    @classmethod
+    def from_json(cls, json):
+        return cls(
+            machine_state=HobStatus.parse_state(json),
+            heater1=HobHeaterStatus.from_json(json, "1")
+            if "Z1status" in json
+            else None,
+            heater2=HobHeaterStatus.from_json(json, "2")
+            if "Z2status" in json
+            else None,
+            heater3=HobHeaterStatus.from_json(json, "3")
+            if "Z3status" in json
+            else None,
+            heater4=HobHeaterStatus.from_json(json, "4")
+            if "Z4status" in json
+            else None,
+            lock=json["lock"] == "1",
+            remote_control=json["StatoWiFi"] == "1",
+        )
+
+    @staticmethod
+    def parse_state(json) -> str:
+        """
+        Parse Hob status
+        """
+        if (
+            json["Z1status"] == "1"
+            or json["Z2status"] == "1"
+            or json["Z3status"] == "1"
+            or json["Z4status"] == "1"
+        ):
+            return HobState.HEATING
+
+        if (
+            json["Z1hot"] == "1"
+            or json["Z2hot"] == "1"
+            or json["Z3hot"] == "1"
+            or json["Z4hot"] == "1"
+        ):
+            return HobState.COOLING_DOWN
+        else:
+            return HobState.IDLE
+
+
+class HoodState(StatusCode):
+    IDLE = (0, "Idle")
+    FAN = (1, "Fan")
+    LIGHT = (2, "Light")
+    FAN_LIGHT = (3, "Fan & Light")
+
+
+@dataclass
+class HoodStatus:
+    machine_state: HoodState
+    light: bool
+    fan: bool
+    grease_filter: bool
+    carbon_Filter: bool
```

Comment:
`carbon_filter`

Reply:
I'.m not shure what you mean with carbon_filter. But original api have property CarbonFilter. So i made property carbon_Filter for store this value.

Reply:
Just that it should be called `carbon_filter` instead of `carbon_Filter`.

Reply:
ok